lang/funcs/core/prometheus/instant_query.go

```diff
@@ -0,0 +1,72 @@
+// Mgmt
+// Copyright (C) 2013-2019+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package coreprometheus
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/purpleidea/mgmt/lang/funcs/simple"
+	"github.com/purpleidea/mgmt/lang/types"
+	"github.com/purpleidea/mgmt/util/errwrap"
+
+	"github.com/prometheus/client_golang/api"
+	"github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+)
+
+func init() {
+	simple.ModuleRegister(ModuleName, "instant_query_value", &types.FuncValue{
+		T: types.NewType("func(query str, ts int, config struct{address str}) str"),
+		V: InstantQueryValue,
+	})
+}
+
+// InstantQuery runs a Prometheus query at a specific time and returns the
```

Comment:
The problem with using this function API, is the value will get pulled at mgmt startup, and then never again. You probably want to use the full function API so that the value can change over time. Whether by polling or ideally by event.

Reply:
That is not how this function works.
Look at the example. It is using datetime.now as parameter.